pypi_mirror.py

```diff
@@ -58,34 +61,94 @@ def are_all_pkgs_downloaded(local_index_path):
 
 
 def update_mirror():
```

Comment:
This whole function probably needs better error-handling in a follow-up PR, because what happens if we run into an error anywhere here? What happens at the next update?

For example, what if some additions/deletions were already sent to RSTUF? Do we send it again?